feature/login/src/main/java/com/epmedu/animeal/login/profile/presentation/FinishProfileScreen.kt

```diff
@@ -0,0 +1,33 @@
+package com.epmedu.animeal.login.profile.presentation
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.epmedu.animeal.extensions.currentOrThrow
+import com.epmedu.animeal.login.presentation.navigateToTabs
+import com.epmedu.animeal.navigation.navigator.LocalNavigator
+
+@Composable
+fun FinishProfileScreen(isFirstTime: Boolean = true) {
+    val viewModel: FinishProfileViewModel = hiltViewModel()
+    val navigator = LocalNavigator.currentOrThrow
+
+    FinishProfileScreenUi(
+        viewModel = viewModel,
+        onBack = navigator::popBackStack
+    )
+
+    LaunchedEffect(Unit) {
+        viewModel.validationEvents.collect {
+            when (it) {
+                FinishProfileViewModel.ValidationEvent.Success -> {
+                    if (isFirstTime) {
```

Comment:
could you please explain why we need this boolean?

Reply:
@egorikftp My suggestion is to use two different *Screen composables with unique navigation calls, using the same *ViewModel and *ScreenUi. In that case we are decoupling the navigation logic (since it lives inside *Screen within callbacks) and sharing the business and ui code, without any booleans.

Reply:
done